src/AtomShell/window.jl

```diff
@@ -31,8 +31,23 @@ function Window(a::Shell, opts::Associative = Dict())
 end
 
 function Window(a::Shell, content::Page, opts::Associative = Dict())
-  opts = merge(opts, Dict(:url => Blink.localurl(content)))
-  return Window(raw_window(a, opts), a, content)
+  url = Blink.localurl(content)
+  is_async(opts) = get(opts, :async, false)
+  callback = !is_async(opts)
+  if callback
+      id, cond = Blink.callback!()
+      url *= "?callback=$id"
```

Comment:
Also, is there a better way to pass the callback ID than through a query param like this? I'm just not very familiar with the framework being used here. I tried to stick it into `req[:params]` like how the page `id` is passed ([here](https://github.com/JunoLab/Blink.jl/pull/137/files#diff-72748641b3d31e360df1094fc47efd08R19)), but i couldn't figure it out...

Reply:
I feel like there should be a better way, but I haven't found it. Good enough for now, imho.

Reply:
👍 Okay cool! I added a TODO comment there in case someone else comes across it and knows a better way! :)